.github/workflows/misc.yml

```
@@ -0,0 +1,14 @@
+name: misc
+on: [push, pull_request]
+jobs:
+  lint:
+    name: miscellaneous checks
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Check for currency.NewPair(BTC, USD) used instead of currency.NewBTCUSD
+      run: |
+        set +e
```

Several improvements can be done here:

```suggestion
      run: |
        grep -r -n -E "currency.NewPair\(currency.BTC, currency.USDT?\)" * || exit 0
        echo "::error::Replace currency.NewPair(BTC, USD*) with currency.NewBTCUSD*()"
        exit 1
```
1) Can use '-n' for line numbers with grep to get more precise output since ATM it would just show the file and matches
2) Can rid the +e as it's not needed
3) Can use the GHA`::error::` macro to replace the current colour eyesore (the colour was green but I'd opt for red on an error and green for success) to be consistent with GHA events

---

Yep, that's better, thank you 🙏 

Fixed https://github.com/gbjk/gocryptotrader/commit/7ab3337117